lib/assessment.rb

```diff
@@ -5,35 +5,61 @@
 # Create a class called Dog. It should have, as attributes, name, breed, and age.
 # Make all of those attributes read-only.
 
-##
-# your code here
-##
+class Dog
+
+    attr_reader :name
+    attr_reader :breed
+    attr_reader :age
+
+    def initialize(name, breed, age)
+        @name = name
+        @breed = breed
+        @age = age
+    end
+end
 
 # Question 2
 # Create a new Dog object.
 
-Answer.dog = nil # replace nil with your code
+Answer.dog = Dog.new("mydog", "Golden Retriever", 15) # replace nil with your code
 
 # Question 3
 # Create a Person class, with attributes name, age, and location.
 # It should set all three attributes when you create a new Person.
 # Name and location should be writeable, but age should be read-only.
 
-##
-# your code here
-##
+class Person
+
+    attr_writer :name
+    attr_reader :age
+    attr_writer :location
```

Comment:
Oops - missed this earlier. Although it's not as semantic, you should really be using `attr_accessor` for `:name` and `:location`, not `attr_writer`. The reason for this is that `attr_writer` creates a setter method _but not a getter method_. This is an extremely unusual pattern; usually, if we can write something, we can read it as well.